languages/python/reference/exercise-concepts/phone-number.md

```diff
@@ -0,0 +1,75 @@
+[Example Solution](https://github.com/exercism/python/blob/master/exercises/phone-number/example.py)
+
+```
+import re
+
+
+class PhoneNumber:
+    def __init__(self, number):
+        self.number = self._clean(number)
+        self.area_code = self.number[:3]
+        self.exchange_code = self.number[3:6]
+        self.subscriber_number = self.number[-4:]
+
+    def pretty(self):
+        return "({}) {}-{}".format(
+            self.area_code, self.exchange_code, self.subscriber_number
+        )
+
+    def _clean(self, number):
+        return self._normalize(re.sub(r"[^\d]", "", number))
+
+    def _normalize(self, number):
+        if len(number) == 10 or len(number) == 11 and number.startswith("1"):
+            valid = number[-10] in "23456789" and number[-7] in "23456789"
+        else:
+            valid = False
+
+        if valid:
+            return number[-10:]
+        else:
+            raise ValueError("{} is not a valid phone number".format(number))
+```
+
+## Concepts
+
+**Classes**
+ - classes are defined with the `class <ClassName>:` syntax
+ - classes require an `__init__` method, whose first argument is `self`, because the result of `__init__` is a class *instance*
+ - classes can have instance *methods* which are called from an instance of the class (as opposed to class methods, called from the Class itself)
+ - classes can define a `property` by using the `@property` decorator (not shown here)
+ - a `property` can be "lazily evaluated" to avoid uneeded computation
+ - "privacy" in Python: Methods prefixed with an underscore, `_`, are conventionally treated as private methods, although Python does not actually support privacy.
```

Comment:
I'm always a little leery of even connecting this idea to "privacy" ... there are "public" methods / attributes (no leading underscore) and there are "non-public" implementation details ... fundamentally the only difference is that the former show up in `help(my_class)` and should always have a docstring. When a lot of people coming from other languages read "privacy" they tend to think that the data is _somehow_ hidden or obfuscated, and there's no facility whatsoever for that in the language.

Reply:
That makes sense. I can change the words to emphasize "convention" and should probably add something along the lines of "can expect non-prefixed methods to stay stable along with semver, and `_` prefixed ones can change at any time" or something similar. i.e. shouldn't import "private" functions or methods from a module, generally.